tests/test_routing.py

```diff
@@ -1164,6 +1162,65 @@ def test_extract_straher_streams_watersheds_d8(self):
         watershed_vector = None
         watershed_layer = None
 
+    def test_single_drain_point(self):
+        """PGP.routing: test single_drain_point pitfill."""
+        dem_array = numpy.zeros((11, 11), dtype=numpy.float32)
+        dem_array[0, 0] = -1.0
+        dem_array[1:8, 1:8] = -2.0
+        dem_array[10, 7] = -4
+        dem_array[8:11, 8:11] = 2.0
+        dem_array[9:11, 9:11] = 1.0
+        dem_array[10, 10] = -7.0
+        dem_path = os.path.join(self.workspace_dir, 'dem.tif')
+        _array_to_raster(dem_array, None, dem_path)
+
+        # outlet tuple at 0,0, just drain one edge
+        expected_array_0_0 = numpy.copy(dem_array)
+        expected_array_0_0[1:8, 1:8] = -1
```

Comment:
Just my own curiosity here. A plateau of all `-1` will drain to an outlet of also `-1`? I guess I'd expect these to be ever so slightly greater than `-1`.

Reply:
By plateau I guess I mean pit in this instance.

Reply:
Yes because there's a -1 pixel that pours off the edge, so to drain you only need the other pixels to be at the same height.